### PR TITLE
AP_Notify: fix incorrect buzzer pin type to stop always instantiating

### PR DESCRIPTION
**Pull request by Hwurzburg**
type and compare is wrong....it always instantiates (unless pin =0 which is not default)

### CODE REVIEW

File: libraries/AP_Notify/Buzzer.cpp [hunk 1]
```diff
@@ -29,7 +29,7 @@ bool Buzzer::init()
         return false;
     }
     _pin = pNotify->get_buzz_pin();
```
Comment (IamPete1) [inline libraries/AP_Notify/Buzzer.cpp:31]: I think you also need to change `get_buzz_pin` to return `int8_t`

Comment (Hwurzburg) [reply libraries/AP_Notify/Buzzer.cpp:31]: duh...yep..done

File: libraries/AP_Notify/Buzzer.cpp [hunk 2]
```diff
@@ -29,7 +29,7 @@ bool Buzzer::init()
...
-    if (_pin == 0u) {
+    if (_pin >= 0) {
```
[the remark this answers is not in the document]
Comment (Hwurzburg) [reply libraries/AP_Notify/Buzzer.cpp]: needed more coffee I guess..thanks

**Review by IamPete1 — APPROVED** [review]
LGTM

**Review by khancyr — APPROVED** [review]

**Review by tridge — APPROVED** [review]

**Comment by rmackay9** [comment]
I've included this in Copter-4.4.0-beta2

**Pull request merged**